daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -378,14 +392,47 @@ object Value extends CidContainer1WithDefaultCidResolver[Value] {
       implicit val `V0 Order`: Order[V0] = Order.fromScalaOrdering[String] contramap (_.coid)
     }
 
-    final case class V1(discriminator: crypto.Hash, suffix: Bytes = Bytes.Empty)
-        extends AbsoluteContractId {
+    final class V1(val discriminator: crypto.Hash, val suffix: Bytes) extends AbsoluteContractId {
```

Comment:
```suggestion
    final class V1 private(val discriminator: crypto.Hash, val suffix: Bytes) extends AbsoluteContractId {
```
Otherwise you don't have the control that the suffix length is respected.

Reply:
good catch.